src/TreeNode.js

```diff
@@ -20,14 +20,6 @@ export default class TreeNode {
      * @constructor
      */
     constructor(options) {
-        // variables for the node
-        this.left = undefined;
-        this.right = undefined;
-        this.distribution = undefined;
-        this.splitValue = undefined;
-        this.splitColumn = undefined;
-        this.gain = undefined;
-
         // options parameters
         this.options = options;
```

Comment:
or access explicity to the property, for me both are valid, but you should choose one